cmd/apiserver_receive_adapter/main.go

```diff
@@ -48,25 +56,54 @@ type envConfig struct {
 	ApiVersion []string `split_words:"true" required:"true"`
 	Kind       []string `required:"true"`
 	Controller []bool   `required:"true"`
+	// MetricsConfigBase64 is a base64 encoded json string of
+	// metrics.ExporterOptions. This is used to configure the metrics exporter
+	// options, the config is stored in a config map inside the controllers
+	// namespace and copied here.
+	MetricsConfigBase64 string `envconfig:"K_METRICS_CONFIG" required:"true"`
+
+	// LoggingConfigBase64 is a base64 encoded json string of logging.Config.
+	// This is used to configure the logging config, the config is stored in
+	// a config map inside the controllers namespace and copied here.
+	LoggingConfigBase64 string `envconfig:"K_LOGGING_CONFIG" required:"true"`
 }
 
 // TODO: the controller should take the list of GVR
 
 func main() {
 	flag.Parse()
 
-	logCfg := zap.NewProductionConfig()
-	logCfg.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
-	dlogger, err := logCfg.Build()
+	var env envConfig
+	err := envconfig.Process("", &env)
 	if err != nil {
-		log.Fatalf("Error building logger: %v", err)
+		panic(fmt.Sprintf("Error processing env var: %s", err))
 	}
-	logger := dlogger.Sugar()
 
-	var env envConfig
-	err = envconfig.Process("", &env)
+	// Convert base64 encoded json logging.Config to logging.Config.
+	loggingConfig, err := resources.Base64ToLoggingConfig(
+		env.LoggingConfigBase64)
 	if err != nil {
-		logger.Fatalw("Error processing environment", zap.Error(err))
+		panic(err)
+	}
+
+	// Convert base64 encoded json metrics.ExporterOptions to
+	// metrics.ExporterOptions.
+	metricsConfig, err := resources.Base64ToMetricsOptions(
+		env.MetricsConfigBase64)
+	if err != nil {
+		panic(err)
+	}
+
+	logger, _ := logging.NewLoggerFromConfig(loggingConfig, component)
```

Comment:
put this above the call to resources.Base64ToMetricsOptions

Reply:
fixed